src/main/java/com/shopify/model/ShopifyProductUpdateRequest.java

```diff
@@ -148,57 +148,55 @@ public PublishedStep withVariantRequests(final List<ShopifyVariantRequest> varia
 				changed = true;
 			}
 
-			final List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
-			final List<Integer> positions = new ArrayList<>(variantRequests.size());
+			List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
+			List<Integer> positions = new ArrayList<>(variantRequests.size());
 
 			for (int i = 0; i < variantRequests.size(); i++) {
 				final ShopifyVariantRequest shopifyVariantRequestForPosition = variantRequests.get(i);
 				positions.add(shopifyVariantRequestForPosition.getRequest().getPosition());
-
 			}
 
-			int maxPosition = variantRequests.stream().map(ShopifyVariantRequest::getRequest)
-					.map(ShopifyVariant::getPosition).max(Comparator.naturalOrder()).get();
+			int maxPosition = positions.stream().max(Comparator.naturalOrder()).get();
+
 			Collections.sort(variantRequests, new ShopifyVariantRequestOption1Comparator());
+
+			updateVariants(variantRequests, shopifyVariants, positions, maxPosition);
+
+			shopifyProduct.setVariants(shopifyVariants);
+
+			return this;
+		}
+
+		private void updateVariants(final List<ShopifyVariantRequest> variantRequests,
+				final List<ShopifyVariant> shopifyVariants,
+				final List<Integer> positions,
+				int maxPosition) {
 			for (int i = 0; i < variantRequests.size(); i++) {
 				final ShopifyVariantRequest shopifyVariantRequest = variantRequests.get(i);
+
 				if (shopifyVariantRequest.hasChanged()) {
 					changed = true;
 				}
 
 				final ShopifyVariant shopifyVariant = shopifyVariantRequest.getRequest();
 
 				if (shopifyVariant.getPosition() == 0) {
-
-					maxPosition = maxPosition + 1;
+					maxPosition += 1;
```

Comment:
It's not the best idea to change variables passed into the function as this changes the reference. Calling code may not know or expect the value they pass in to be manipulated so it's a good idea to avoid changing a value of a method argument.

In this case, you don't need to pass maxPosition as an object to this function. It can be calculated right in this function directly because it's not used any where else and you don't have to worry about this. It also cuts down an argument needed to call this function.